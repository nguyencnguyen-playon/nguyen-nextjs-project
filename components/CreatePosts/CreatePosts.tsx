'use client';
import { useRouter } from 'next/navigation';
import { LayoutContent } from '@/components/LayoutContent';
import styled from 'styled-components';
import { Button, Form, Input } from 'antd';
import { PostFormType } from '@/interfaces';
import { createPosts } from '@/services';

const Wrapper = styled.div`
    display: flex;  
    flex:1;
    flex-wrap: wrap;
    justify-content: center;
    background: #F7FAFC;
    overflow: auto;
    margin-top: 32px;
`

export const CreatePosts = () => {
    const router = useRouter();

    const handleCreate = async (data: PostFormType) => {
        await createPosts(data);
        router.push('/');
    };

    return (
        <LayoutContent>
            <Wrapper>
                <Form
                    name="basic"
                    labelCol={{ span: 8 }}
                    wrapperCol={{ span: 16 }}
                    style={{ maxWidth: 600 }}
                    initialValues={{ remember: true }}
                    onFinish={handleCreate}
                    autoComplete="off"
                >
                    <Form.Item<PostFormType>
                        label="Username"
                        name="name"
                        rules={[{ required: true, message: 'Please input your username!' }]}
                    >
                        <Input />
                    </Form.Item>

                    <Form.Item<PostFormType>
                        label="Description"
                        name="desc"
                        rules={[{ required: true, message: 'Please input your description!' }]}
                    >
                        <Input />
                    </Form.Item>

                    <Form.Item wrapperCol={{ offset: 8, span: 16 }}>
                        <Button type="primary" htmlType="submit">
                            Submit
                        </Button>
                    </Form.Item>
                </Form>
            </Wrapper>
        </LayoutContent>
    );
};

