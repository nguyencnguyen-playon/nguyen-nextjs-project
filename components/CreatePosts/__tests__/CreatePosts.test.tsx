import { render } from '@testing-library/react';
import { CreatePosts } from '../CreatePosts';

Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
    })),
});

jest.mock('next/navigation', () => ({
    ...jest.requireActual('next/navigation'),
    useRouter: jest.fn()
}));

jest.mock('@clerk/nextjs', () => ({
    ...jest.requireActual('@clerk/nextjs'),
    useUser: jest.fn().mockImplementation(() => ({
        user: null
    })),
    useClerk: jest.fn().mockImplementation(() => ({
        signOut: jest.fn()
    })),
    SignedIn: () => <div />,
    UserButton: () => <div />
}));

describe('CreatePosts', () => {
    test('renders without errors', () => {
        const component = render(<CreatePosts />);
        expect(component).toMatchSnapshot();
    });
});
