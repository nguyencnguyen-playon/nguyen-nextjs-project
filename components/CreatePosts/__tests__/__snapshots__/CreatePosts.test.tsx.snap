// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePosts renders without errors 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-beqWNU boLFlS"
      >
        <span
          class="sc-gueXAH dbBicJ"
        >
          Hello 
        </span>
        <div />
        <div
          class="sc-dmqGRC jNxXzf"
        >
          <a
            class="sc-hLseSp eSJDNf"
            href="/"
          >
            Home
          </a>
          <a
            class="sc-hLseSp eSJDNf"
            href="/sign-in"
          >
            Sign In
          </a>
        </div>
      </header>
      <div
        class="sc-jTrOVX cicrdZ"
      >
        <div
          class="sc-gLDymS bGitmd"
        >
          <form
            autocomplete="off"
            class="ant-form ant-form-horizontal css-dev-only-do-not-override-1okl62o"
            id="basic"
            style="max-width: 600px;"
          >
            <div
              class="ant-form-item css-dev-only-do-not-override-1okl62o"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-1okl62o"
              >
                <div
                  class="ant-col ant-col-8 ant-form-item-label css-dev-only-do-not-override-1okl62o"
                >
                  <label
                    class="ant-form-item-required"
                    for="basic_name"
                    title="Username"
                  >
                    Username
                  </label>
                </div>
                <div
                  class="ant-col ant-col-16 ant-form-item-control css-dev-only-do-not-override-1okl62o"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        aria-required="true"
                        class="ant-input css-dev-only-do-not-override-1okl62o ant-input-outlined"
                        id="basic_name"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item css-dev-only-do-not-override-1okl62o"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-1okl62o"
              >
                <div
                  class="ant-col ant-col-8 ant-form-item-label css-dev-only-do-not-override-1okl62o"
                >
                  <label
                    class="ant-form-item-required"
                    for="basic_desc"
                    title="Description"
                  >
                    Description
                  </label>
                </div>
                <div
                  class="ant-col ant-col-16 ant-form-item-control css-dev-only-do-not-override-1okl62o"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        aria-required="true"
                        class="ant-input css-dev-only-do-not-override-1okl62o ant-input-outlined"
                        id="basic_desc"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item css-dev-only-do-not-override-1okl62o"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-1okl62o"
              >
                <div
                  class="ant-col ant-col-16 ant-col-offset-8 ant-form-item-control css-dev-only-do-not-override-1okl62o"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <button
                        class="ant-btn css-dev-only-do-not-override-1okl62o ant-btn-primary"
                        type="submit"
                      >
                        <span>
                          Submit
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-beqWNU boLFlS"
    >
      <span
        class="sc-gueXAH dbBicJ"
      >
        Hello 
      </span>
      <div />
      <div
        class="sc-dmqGRC jNxXzf"
      >
        <a
          class="sc-hLseSp eSJDNf"
          href="/"
        >
          Home
        </a>
        <a
          class="sc-hLseSp eSJDNf"
          href="/sign-in"
        >
          Sign In
        </a>
      </div>
    </header>
    <div
      class="sc-jTrOVX cicrdZ"
    >
      <div
        class="sc-gLDymS bGitmd"
      >
        <form
          autocomplete="off"
          class="ant-form ant-form-horizontal css-dev-only-do-not-override-1okl62o"
          id="basic"
          style="max-width: 600px;"
        >
          <div
            class="ant-form-item css-dev-only-do-not-override-1okl62o"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1okl62o"
            >
              <div
                class="ant-col ant-col-8 ant-form-item-label css-dev-only-do-not-override-1okl62o"
              >
                <label
                  class="ant-form-item-required"
                  for="basic_name"
                  title="Username"
                >
                  Username
                </label>
              </div>
              <div
                class="ant-col ant-col-16 ant-form-item-control css-dev-only-do-not-override-1okl62o"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      aria-required="true"
                      class="ant-input css-dev-only-do-not-override-1okl62o ant-input-outlined"
                      id="basic_name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item css-dev-only-do-not-override-1okl62o"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1okl62o"
            >
              <div
                class="ant-col ant-col-8 ant-form-item-label css-dev-only-do-not-override-1okl62o"
              >
                <label
                  class="ant-form-item-required"
                  for="basic_desc"
                  title="Description"
                >
                  Description
                </label>
              </div>
              <div
                class="ant-col ant-col-16 ant-form-item-control css-dev-only-do-not-override-1okl62o"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      aria-required="true"
                      class="ant-input css-dev-only-do-not-override-1okl62o ant-input-outlined"
                      id="basic_desc"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-form-item css-dev-only-do-not-override-1okl62o"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-1okl62o"
            >
              <div
                class="ant-col ant-col-16 ant-col-offset-8 ant-form-item-control css-dev-only-do-not-override-1okl62o"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-1okl62o ant-btn-primary"
                      type="submit"
                    >
                      <span>
                        Submit
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
